lib/ui/common/CircularProgress.dart

```diff
@@ -102,22 +33,67 @@ class _CircularProgressState extends State<CircularProgress> {
   Widget build(BuildContext context) {
     return Center(
       child: Container(
-        height: widget._style.height,
-        width: widget._style.width,
+        height: height,
+        width: width,
         child: CustomPaint(
-          foregroundPainter: CircularPainter(
-              lineColor: widget._style.lineColor,
-              completeColor: widget._style.completeColor,
-              completePercent: widget._viewModel.initialValue,
-              width: widget._style.lineWidth),
+          foregroundPainter: _CircularPainter(
+              lineColor: lineColor,
+              completeColor: widget._lineColor,
+              completePercent: widget._progress,
+              width: lineWidth),
           child: Padding(
-            padding: widget._style.innerPadding,
-            child: ClipOval(
-              child: Stack(children: widget._viewModel.content),
-            ),
+            padding: innerPadding,
           ),
         ),
       ),
     );
   }
 }
+
+class _CircularPainter extends CustomPainter{
+  Color lineColor;
+  Color completeColor;
+  double completePercent;
+  double width;
+
+
+  _CircularPainter({this.lineColor, this.completeColor, this.completePercent,
+    this.width});
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    Paint line = Paint()
+      ..color = lineColor
+      ..strokeCap = StrokeCap.round
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = width;
+    Paint complete = Paint()
+      ..color = completeColor
+      ..strokeCap = StrokeCap.round
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = width;
+
+    Offset center = Offset(size.width/2, size.height/2);
+    double radius = min(size.width/2, size.height/2);
+    canvas.drawCircle(
+        center,
+        radius,
+        line
+    );
+
+    double arcAngle = 2*pi* (completePercent/100);
```

Comment:
take in a normalised number, not per cent. that removes this divide. clamp the input to 0.0-1.0 so it will never bet outside this range here.